transformer_engine/pytorch/module/grouped_linear.py

```diff
@@ -536,23 +518,6 @@ class GroupedLinear(TransformerEngineBaseModule):
           responsibility to ensure all parameters are moved to the GPU before running the
           forward pass.
 
-    Parallelism parameters
-    ----------------------
-    sequence_parallel : bool, default = `False`
-                       if set to `True`, uses sequence parallelism.
-    tp_group : ProcessGroup, default = `None`
-              tensor parallel process group.
-    tp_size : int, default = 1
-             used as TP (tensor parallel) world size when TP groups are not formed during
-             initialization. In this case, users must call the
-             `set_tensor_parallel_group(tp_group)` method on the initialized module before the
-             forward pass to supply the tensor parallel group needed for tensor and sequence
-             parallel collectives.
-    parallel_mode : {None, 'column', 'row'}, default = `None`
-                   used to decide whether this GroupedLinear layer is Column Parallel Linear or Row
-                   Parallel Linear as described `here <https://arxiv.org/pdf/1909.08053.pdf>`_.
-                   When set to `None`, no communication is performed.
-
```

Comment:
We can remove them from `_GroupedLinear`, but we still need them in the `GroupedLinear` as they are used, right? In this case we should retain the docs.

Reply:
Well, that's the point - they are not actually used in the code and do not really work (e.g. there is no communication happening), so I do not want people to think that they can use them.
I need to keep the actual parameters in the constructor to not break MCore but at the very least do not want to advertise functionality that is not there.

Reply:
We could set the kwarg defaults to `None` and print a warning if they are set.

Reply:
Agreed, but at this point I want to just minimize the exposure rather than properly fix this (also I do not know what exactly is passed to those modules inside MCore, so there is a possibility that such warning would be triggered by them even if they are otherwise fine).